### PR TITLE
Add BayesUCB to DDPG and accelerated the speed of learning

### PR DESCRIPTION
**Pull request by toobrainless**
Old version: 

DDPG_res_1

New version:

DDPG_res_2

### CODE REVIEW

File: replay/models/ddpg.py [hunk 1]
```diff
@@ -444,12 +537,18 @@ def __init__(
...
         noise_sigma: float = 0.2,
         noise_theta: float = 0.05,
-        noise_type: str = "ou",
+        noise_type: str = "gauss",
         seed: int = 9,
         user_num: int = 10,
         item_num: int = 10,
         log_dir: str = "logs/tmp",
         exact_embeddings_size=True,
+        n_critics_head: int = 10,
```
Comment (shashist) [inline replay/models/ddpg.py:546]: could you please update descriptions? e.g. add link to BayesUCBDQN paper, indicate which arguments are related to it

File: replay/models/ddpg.py [hunk 2]
```diff
@@ -504,13 +622,19 @@ def _batch_pass(self, batch: dict) -> Dict[str, Any]:
...
+            reward + (1.0 - done) * self.gamma * target_value
+        )  # smth strange, check article
         expected_value = torch.clamp(
             expected_value, self.min_value, self.max_value
         )
 
         value = self.value_net(state, action)
-        value_loss = (value - expected_value.detach()).squeeze(1).pow(2).mean()
+        value_loss = (
+            (((value - expected_value.detach())).pow(2) * sample_weight)
```
Comment (shashist) [inline replay/models/ddpg.py:634]: is this weighting related to BayesUCBDQN?

Comment (toobrainless) [reply replay/models/ddpg.py:634]: Yes

File: replay/models/ddpg.py [hunk 3]
```diff
@@ -627,7 +750,7 @@ def _preprocess_log(log):
 
     def _get_batch(self, step: int = 0) -> dict:
         beta = self._get_beta(step)
```
Comment (shashist) [inline replay/models/ddpg.py:752]: do we still need this line (and self._get_beta function)?

**Comment by shashist** [comment]
also update corresponding tests, please

**Comment by shashist** [comment]
Overall code looks nice. Probably final comment: we want keep python 3.7 compatibility, so please make sure tests are passed on 3.7 version. Thank you

**Review by shashist — APPROVED** [review]
